discovery-service/src/main/java/org/zowe/apiml/discovery/ApimlInstanceRegistry.java

```diff
@@ -190,4 +201,40 @@ public boolean statusUpdate(String appName, String instanceId, InstanceInfo.Inst
         this.appCntx.publishEvent(new EurekaStatusUpdateEvent(this, appName, instanceId));
         return isUpdated;
     }
+
+    /**
+     * Change the service ID prefix according to the regex before the service registers to Eureka.
+     * @param info the instance info
+     * @param regex the regex
+     * @return instance info with the modified service ID
+     */
+    protected InstanceInfo changeServiceId(final InstanceInfo info, String regex) {
+        String servicePrefix = regex.split(",")[0];
+        String targetValue = regex.split(",")[1];
+        String instanceId = info.getInstanceId();
+        if (instanceId.contains(servicePrefix)) {
+            String appName = info.getAppName();
+            instanceId = instanceId.replace(servicePrefix, targetValue);
+            appName = appName.replace(servicePrefix.toUpperCase(), targetValue.toUpperCase());
+            log.debug("The instance ID of {} service has been changed to {}.", info.getAppName(), instanceId);
+            return new InstanceInfo.Builder(info)
+                .setInstanceId(instanceId)
+                .setAppGroupName(appName)
+                .setAppName(appName)
+                .setVIPAddress(appName)
+                .build();
+        }
+        return info;
+    }
+
+    private boolean isValidRegex(String regex) {
+        String servicePrefix = regex.split(",")[0];
+        String targetValue = regex.split(",")[1];
```

Comment:
what if regex contains string without ","?

Reply:
I will add the check, thanks